src/video/SDL_video.c

```diff
@@ -2060,21 +2060,22 @@ int SDL_SetWindowIcon(SDL_Window *window, SDL_Surface *icon)
     CHECK_WINDOW_MAGIC(window, -1);
 
     if (icon == NULL) {
-        return 0;
+        return SDL_SetError("SDL_SetWindowIcon(): Invalid surface");
     }
 
     SDL_DestroySurface(window->icon);
 
     /* Convert the icon into ARGB8888 */
     window->icon = SDL_ConvertSurfaceFormat(icon, SDL_PIXELFORMAT_ARGB8888);
     if (!window->icon) {
-        return -1;
+        return SDL_SetError("Unable to convert the icon's format");
```

Comment:
SDL_ConvertSurfaceFormat() will provide an error message, this additional error isn't needed.